elements/lisk-p2p/src/peer/base.ts

```diff
@@ -174,6 +191,22 @@ export class Peer extends EventEmitter {
 				return;
 			}
 
+			// Apply penalty when you receive getNodeInfo RPC more than once
+			if (rawRequest.procedure === REMOTE_EVENT_RPC_GET_NODE_INFO) {
+				if (this._discoveryMessageCounter.getNodeInfo >= 1) {
```

Comment:
I think this should compare with the rate (ie until 4th time, it shouldnt apply penalty?
```
				if (this._discoveryMessageCounter.getNodeInfo > this._peerStatusMessageRate) {
```

Reply:
_peerStatusMessageRate is for `postNodeInfo` i.e., only for status messages. whereas for RPC requests a peer is only allowed to request nodeInfo or peerlist once when it makes a connection.
`Add RPC requests counter for getPeers and getNodeInfo, count > 1 => applyPenalty(10)`